nanoFramework.System.Net/Sockets/Socket.cs

```diff
@@ -336,39 +346,69 @@ public void Bind(EndPoint localEP)
         /// </remarks>
         public void Connect(EndPoint remoteEP)
         {
-            if (m_Handle == -1)
+            try
```

Comment:
Similar comment. This is supposed to throw exceptions when something goes wrong. See [here](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socket.connect?view=netframework-4.8#System_Net_Sockets_Socket_Connect_System_Net_EndPoint_)

Again: we can hide those, it's up to the caller to handle them.
Otherwise we are are breaking the compatibility/behaviour with the full .NET framework.